### PR TITLE
51 city information

### PR DESCRIPTION
**Pull request by reemsh2000**
* city information section

### CODE REVIEW

File: client/src/components/city.css [hunk 1]
```diff
@@ -0,0 +1,85 @@
+.city-info-container{
+  position: relative;
+  width:90%;
+  padding-top: 120px;
```
Comment (Nizar7zak) [inline client/src/components/city.css:4]: why you use padding top here?

**Review by Nizar7zak — CHANGES_REQUESTED** [review]
Great Job

**Review by modrayyes — APPROVED** [review]

File: client/src/components/CityInformation.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  const [families, setfamilies] = useState([]);
+  useEffect(() => {
+    const getCityData = async () => {
+      const cityInformation = await httpService.get(`/api/v1/city/${id}`);
+      const cityFamilies = await httpService.get(`/api/v1/families/${id}`);
+      const { cityData } = cityInformation.data.data;
+      setcityInfo(cityData);
+      setfamilies(cityFamilies);
+    };
+    getCityData();
```
Comment (LinaYahya) [inline client/src/components/CityInformation.jsx:22]: cleanup our effect!

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/components/CityInformation.jsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable no-console */
```
Comment (LinaYahya) [inline client/src/components/CityInformation.jsx:2]: remove these lines, and fix those rules

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/components/CityInformation.jsx (hunk 3 above)
Comment (reemsh2000) [reply client/src/components/CityInformation.jsx:2]: done

File: client/src/pages/City.jsx [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
+import CityInformation from '../components/CityInformation';
 
 function City() {
   return (
-    <div />
+    <CityInformation id={1} />
```
[the remark this answers is not in the document]
Comment (reemsh2000) [reply client/src/pages/City.jsx:6]: done

**Review by NadiaJalalTamimi — APPROVED** [review]

**Review by Nizar7zak — APPROVED** [review]

File: client/src/components/CityInformation.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from 'react';
```
Comment (LinaYahya) [inline client/src/components/CityInformation.jsx:1]: make a folder for city

File: client/src/components/CityInformation.jsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import Img from './common/Img';
+import ParagraphText from './common/Paragraph';
+import HeaderTitle from './common/Title';
+import './city.css';
+
+function CityInformation() {
+  const { id } = useParams();
+  const [cityInfo, setcityInfo] = useState({});
+  const [families, setfamilies] = useState([]);
+  useEffect(() => {
```
Comment (LinaYahya) [inline client/src/components/CityInformation.jsx:15]: we'll fetch data from the city page, check cityProvider and get data from there

File: client/src/context/AuthContext.jsx [hunk 7]
```diff
@@ -1,17 +1,28 @@
...
-      const data = await httpService.get('/api/v1/userinfo');
-      setUserData(data);
+    try {
+      const getData = async () => {
+        const data = await http.get('/api/v1/userinfo');
+        setUserData(data);
+      };
+      getData();
+    } catch (error) {
+      if (error.status >= 400 && error.status <= 500) {
```
Comment (LinaYahya) [inline client/src/context/AuthContext.jsx:18]: no need for this condition

File: client/src/context/AuthContext.jsx [hunk 8]
```diff
@@ -1,17 +1,28 @@
...
+        message.error(error.message);
```
Comment (LinaYahya) [inline client/src/context/AuthContext.jsx:19]: no need to this, just setUser to null

**Review by LinaYahya — CHANGES_REQUESTED** [review]
Put your images in utils/image folder in the src and import it in the file you use

File: client/src/context/AuthContext.jsx (hunk 7 above)
Comment (reemsh2000) [reply client/src/context/AuthContext.jsx:18]: done

File: client/src/context/AuthContext.jsx (hunk 8 above)
Comment (reemsh2000) [reply client/src/context/AuthContext.jsx:19]: done

File: client/src/components/CityInformation.jsx (hunk 5 above)
Comment (reemsh2000) [reply client/src/components/CityInformation.jsx]: done

File: client/src/components/CityInformation.jsx (hunk 6 above)
Comment (reemsh2000) [reply client/src/components/CityInformation.jsx]: done

**Pull request merged**